x/concentrated-liquidity/fuzz_test.go

```diff
@@ -278,6 +278,9 @@ func (s *KeeperTestSuite) swap(pool types.ConcentratedPoolExtension, swapInFunde
 	// // Execute swap
 	fmt.Printf("swap in: %s\n", swapInFunded)
 	cacheCtx, writeOutGivenIn := s.Ctx.CacheContext()
+	pool, err := s.clk.GetPoolById(s.Ctx, pool.GetId())
+	s.Require().NoError(err)
```

Comment:
Why is this needed in place of the pool object that is provided in the args? No state change occurs prior to this call on the pool obj as far as I see.

Reply:
So pool obj change does when we create first position. It updates the current Sqr Price https://github.com/osmosis-labs/osmosis/blob/main/x/concentrated-liquidity/lp.go#L476